csvcubed/csvcubed/readers/cubeconfig/v1/columnschema.py

```diff
@@ -78,25 +92,66 @@ def map_to_new_qb_dimension(
         )
         # The NewQbCodeList and Concepts are populated in the NewQbDimension.from_data() call
         # the _get_code_list method overrides the code_list if required.
-        new_dimension.code_list = self._get_code_list(new_dimension, csv_column_title)
-        return new_dimension
+        (
+            new_dimension.code_list,
+            code_list_schema_validation_errors,
+        ) = self._get_code_list(
+            new_dimension,
+            csv_column_title,
+            cube_config_minor_version,
+            cube_config_path=config_path,
+        )
+        return (new_dimension, code_list_schema_validation_errors)
 
     def _get_code_list(
-        self, new_dimension: NewQbDimension, csv_column_title: str
-    ) -> Optional[QbCodeList]:
-
+        self,
+        new_dimension: NewQbDimension,
+        csv_column_title: str,
+        cube_config_minor_version: Optional[int],
+        cube_config_path: Optional[Path],
+    ) -> Tuple[Optional[QbCodeList], Optional[list[ValidationError]]]:
```

Comment:
I don't tend to use `Optional` with lists since an empty list usually conveys the same information.